### PR TITLE
Applying new tables to [reviews] page.

### PR DESCRIPTION
**Pull request by joanofdart**
1 gone, some to go!

### CODE REVIEW

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 1]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
+
+            <div fxLayout fxLayoutGap="24px">
+              <mat-form-field appearance="outline" fxFlex>
+                <mat-label>Producer's Name</mat-label>
+                <input test-id="first-name" matInput formControlName="firstName" required />
+              </mat-form-field>
+
+              <mat-form-field appearance="outline" fxFlex>
+                <mat-label>Producer's Lastname</mat-label>
+                <input test-id="first-name" matInput formControlName="lastName" required />
```
Comment (fritzschoff) [inline libs/movie/src/lib/movie/pages/production/production.component.html:62]: same test id then above

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 2]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-          <span class="mat-caption">Delete Distributor</span>
-        </footer>
-      </article>
-      <footer fxLayout="column" fxLayoutAlign="center center">
-        <button mat-mini-fab color="primary" (click)="distributors.add()" aria-label="Remove a distributor">
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Distributor</span>
-      </footer>
+      <bf-form-table [columns]="displayedColumnsDistributors" [form]="distributors" buttonText="Add new Director">
```
Comment (fritzschoff) [inline libs/movie/src/lib/movie/pages/production/production.component.html:77]: button Text is wrong

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 3]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-        </footer>
-      </article>
-      <footer fxLayout="column" fxLayoutAlign="center center">
-        <button mat-mini-fab color="primary" (click)="coProducer.add()" aria-label="Remove a co-company">
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Co-Production Company</span>
-      </footer>
+      <h3>Co-Production Company</h3>
+      <bf-form-table [columns]="displayedColumnsCoProductionCompany" [form]="coProductionCompany" buttonText="Add Production Company">
```
Comment (fritzschoff) [inline libs/movie/src/lib/movie/pages/production/production.component.html:28]: add a CO production company? dunno

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 4]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-          <span class="mat-caption">Delete Sales Company</span>
-        </footer>
-      </article>
-      <footer fxLayout="column" fxLayoutAlign="center center">
-        <button mat-mini-fab color="primary" (click)="internationalSales.add()" aria-label="Remove a sales company">
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Sales Company</span>
-      </footer>
+      <bf-form-table [columns]="displayedColumnsInternationalSales" [form]="internationalSales" buttonText="Add Producer">
```
Comment (fritzschoff) [inline libs/movie/src/lib/movie/pages/production/production.component.html:99]: button text wrong

File: libs/movie/src/lib/movie/pages/production/production.component.ts [hunk 5]
```diff
@@ -10,6 +10,32 @@ import { MovieFormShellComponent } from '../shell/shell.component';
 export class MovieFormProductionComponent {
   form = this.shell.form;
 
+  public displayedColumnsProductionCompany = {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/production/production.component.ts:13]: Maybe change the name for 
- `productionColumns`
- `coProductionColumns`
- `producerColumns`
....

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 6]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-      </article>
-      <footer fxLayout="column" fxLayoutAlign="center center">
-        <button mat-mini-fab color="primary" (click)="executiveProducer.add()" aria-label="Remove a company">
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Production Company</span>
-      </footer>
+      <bf-form-table [columns]="displayedColumnsProductionCompany" [form]="productionCompany" buttonText="Add Production Company">
+        <ng-template formView let-form>
+          <article [formGroup]="form">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/production/production.component.html:8]: I don't think you want the global form here. You want a local version of the productionCompany.
As far as I remember this form is only used locally to update the real form.
Am I correct @fritzschoff  ?

Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/pages/production/production.component.html:8]: Same will go for everything below.

Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/production/production.component.html:8]: So it should be "productionCompany"? afaik this is how the main-component is using it

Comment (fritzschoff) [reply libs/movie/src/lib/movie/pages/production/production.component.html:8]: form comes from `let-form` so all good here. maybe rename it if @GrandSchtroumpf misread this

Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/pages/production/production.component.html:8]: indeed, maybe "localForm" :
```html
<ng-template formView let-localForm>
  <article [formGroup]="localForm"> ... </article>
</ng-template>

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 7]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Co-Production Company</span>
-      </footer>
+      <h3>Co-Production Company</h3>
+      <bf-form-table [columns]="displayedColumnsCoProductionCompany" [form]="coProductionCompany" buttonText="Add Co-Production Company">
+        <ng-template formView let-form>
+          <article [formGroup]="form">
+
+            <div fxLayout fxLayoutGap="24px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/production/production.component.html:32]: I think you don't need this div, put the fxLayout on the article instead.

File: libs/movie/src/lib/movie/pages/production/production.component.html [hunk 8]
```diff
@@ -3,152 +3,115 @@ <h1 subtitle>Production Information</h1>
...
-        <button mat-mini-fab color="primary" (click)="producers.add()" aria-label="Remove a producer">
-          <mat-icon svgIcon="add"></mat-icon>
-        </button>
-        <span class="mat-caption">Add Producer</span>
-      </footer>
+      <bf-form-table [columns]="displayedColumnsProducers" [form]="producers" buttonText="Add Producer">
+        <ng-template formView let-form>
+          <article [formGroup]="form">
+
+            <div fxLayout fxLayoutGap="24px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/production/production.component.html:54]: Same as above, I think you can put the fxLayout on the article. Apply also for below.

**Pull request merged**